src/test/java/org/opensearch/security/ssl/SecuritySSLReloadCertsActionTests.java

```diff
@@ -251,6 +189,117 @@ public void testSSLReloadFail_NoReloadSet() throws Exception {
         Assert.assertEquals(expectedResponseString, reloadCertsResponse.getBody());
     }
 
+    @Test
+    public void testReloadTransportSSLSameCertsPass() throws Exception {
+        initClusterWithTestCerts();
+        RestHelper rh = getRestHelperAdminUser();
+
+        String certDetailsResponse = rh.executeSimpleRequest(GET_CERT_DETAILS_ENDPOINT);
+
+        JSONObject expectedJsonResponse = getInitCertDetailsExpectedResponse();
+        Assert.assertEquals(expectedJsonResponse.toString(), certDetailsResponse);
+
+        // Test Valid Case: Reload same certificate
+        updateFiles(defaultCertFilePath, pemCertFilePath);
+        updateFiles(defaultKeyFilePath, pemKeyFilePath);
+
+        Assert.assertTrue(reloadAndVerifyResponse(rh, "transport", getInitCertDetailsExpectedResponse()));
+    }
+
+    @Test
+    public void testReloadHttpSSLSameCertsPass() throws Exception {
+        initClusterWithTestCerts();
+        RestHelper rh = getRestHelperAdminUser();
+
+        String certDetailsResponse = rh.executeSimpleRequest(GET_CERT_DETAILS_ENDPOINT);
+        JSONObject expectedJsonResponse = getInitCertDetailsExpectedResponse();
+        Assert.assertEquals(expectedJsonResponse.toString(), certDetailsResponse);
+
+        // Test Valid Case: Reload same certificate
+        updateFiles(defaultCertFilePath, pemCertFilePath);
+        updateFiles(defaultKeyFilePath, pemKeyFilePath);
+
+        Assert.assertTrue(reloadAndVerifyResponse(rh, "http", getInitCertDetailsExpectedResponse()));
+    }
+
+    /**
+     *
+     * @param rh RestHelper to perform rest actions on the cluster
+     * @param updateChannel certType/channel being updated, either http or transport
+     * @param expectedCertResponse expected Certs after reload
+     * @return True if all assertions pass
+     * @throws Exception if rest api failed
+     */
+    private Boolean reloadAndVerifyResponse(RestHelper rh, String updateChannel, JSONObject expectedCertResponse) throws Exception {
```

Comment:
What do you think about renaming this method `assertThatReloadCertificates(...)`, then you don't have to pass a bool out, just make it void

Reply:
Sure, updated to assertReloadCertificateSuccess(...)